script.js

```diff
@@ -1,12 +1,247 @@
+        updateEpisodeSelect(allEpisodes);
+      })
+      .catch((error) => console.error(error));
+  });
+
+  shows.forEach((show) => {
+    const option = document.createElement("option");
+    option.value = show.id;
+    option.textContent = show.name;
+    showSelect.appendChild(option);
+  });
+
+  showContainer.appendChild(showSelect);
+  searchSelectDiv.appendChild(showContainer);
+}
+
+//updating the select field to get episodes from all shows
+function updateEpisodeSelect(episodes) {
+  const episodeSelect = document.querySelector(".episode-select");
+  episodeSelect.innerHTML = "";
+
+  const selectOption = document.createElement("option");
+  selectOption.value = "select-episode";
+  selectOption.textContent = "Select Episode";
+  episodeSelect.appendChild(selectOption);
+
+  episodes.forEach((episode) => {
+    const option = document.createElement("option");
+    option.value = episode.id;
+    option.textContent = `${episode.name} - S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")}`;
+    episodeSelect.appendChild(option);
+  });
+}
+
+
+
+//Creating the header of page
+function createHeader() {
+  const body = document.body;
+
+  // create the header element
+  const header = document.createElement("header");
+  header.classList.add("header");
+  body.insertBefore(header, body.firstChild);
+
+  // create the title div
+  const titleHeader = document.createElement("h1");
+  titleHeader.classList.add("title");
+  titleHeader.textContent = "Cineflix";
+  header.appendChild(titleHeader);
+
+  // create the search and select div
+  const searchSelectDiv = document.createElement("div");
+  searchSelectDiv.classList.add("search-select");
+  header.appendChild(searchSelectDiv);
+}
+
+//creating footer 
+function createFooter() {
+  const body = document.body;
+
+  let footer = document.createElement("footer");
+  footer.classList.add("footer");
+
+  // create the attribution
+  let licensing = document.createElement("p");
+  licensing.innerHTML = `This data has (originally) come from <a href="https://www.tvmaze.com/">TVMaze.com</a>`;
+  footer.appendChild(licensing);
+
+  // append the footer to the body
+  body.appendChild(footer);
+}
+
+//level 300-adding an episode selector 
+function episodeSelectField(episodes) {
+  const searchSelectDiv = document.querySelector(".search-select");
+
+  const episodeSelectContainer = document.createElement("div");
+  episodeSelectContainer.classList.add("episode-select-container");
+
+  const episodeSelect = document.createElement("select");
+  episodeSelect.classList.add("episode-select");
+  episodeSelect.addEventListener("change", () => {
+    const episodeId = episodeSelect.value;
+    if (episodeId === "select-episode") {
+      makePageForEpisodes(allEpisodes);
+    } else {
+      const selectedEpisode = allEpisodes.find((episode) => episode.id == episodeId);
+      makePageForEpisodes([selectedEpisode]);
```

Comment:
I think that this selector is meant to scroll down to the episode rather than filter the list of episodes being shown

Reply:
otherwise, you should also update the "Displaying ?/? episodes" text as well

Reply:
i will have a look at this too , i haven't come across it before